Lumpen Radio/Radio.swift

```diff
@@ -56,32 +60,87 @@ class Radio: ObservableObject {
     // Reference:
     // https://stackoverflow.com/questions/48555049/how-to-stream-audio-from-url-without-downloading-mp3-file-on-device
     func startRadio() {
+        if self.player != nil {
+            self.player?.play()
+            self.isPlaying = true
+            self.bottomText = BOTTOM_TEXT_RADIO_ON
+            return
+        }
         guard let url = URL(string: LUMPEN_AUDIO_URL) else {
             print("Could not start radio")
             return
         }
-        let playerItem = AVPlayerItem(url: url)
+        self.playerItem = AVPlayerItem(url: url)
         let player = AVPlayer(playerItem: playerItem)
-        let audioSession = AVAudioSession.sharedInstance()
+        
+        player.play()
+        self.isPlaying = true
+        self.bottomText = BOTTOM_TEXT_RADIO_ON
+        self.player = player    }
+    
+    func stopRadio() {
+        self.player?.pause()
+        // no need to release the player, as it will cause a delay when tuning in again
+//        self.player = nil
+        self.isPlaying = false
+        self.bottomText = BOTTOM_TEXT_RADIO_OFF
+    }
+    
+    func setupAudioSession() {
         do {
-            try audioSession.setCategory(.playback,
-                                         mode: .default,
-                                         options: [.mixWithOthers, .allowAirPlay])
-            try audioSession.setActive(true)
-            player.play()
-            self.isPlaying = true
-            self.bottomText = BOTTOM_TEXT_RADIO_ON
-            self.player = player
+            try AVAudioSession.sharedInstance().setCategory(.playback, mode: .default, options: [.allowBluetooth, .allowAirPlay])
+            try AVAudioSession.sharedInstance().setActive(true)
         } catch {
             print(error)
         }
     }
     
-    func stopRadio() {
-        self.player?.pause()
-        self.player = nil
-        self.isPlaying = false
-        self.bottomText = BOTTOM_TEXT_RADIO_OFF
+    func setupNowPlaying() {
+        // Define Now Playing Info
+        var nowPlayingInfo = [String : Any]()
+        nowPlayingInfo[MPMediaItemPropertyTitle] = "Lumpen Radio - Now Playing"
```

Comment:
Can make the value, "Lumpen Radio - Now Playing", a constant `let` like the others at the top of this file

Reply:
Done